coalib/parsing/LineParser.py

```diff
@@ -30,7 +30,9 @@ def __init__(self,
                                             "section" despite of the current
                                             section.
         """
-        section_name_surroundings = section_name_surroundings or {"[": "]"}
+        section_name_surroundings = (
+            {"[": "]"} if section_name_surroundings is None
+            else section_name_surroundings)
```

Comment:
ah yeah makes sense. The possibility would here to still use `MappingProxyType` and do then:
```
section_name_surroundings = dics(section_name_surrounding)
```
which does a (shallow) copy of the elements.

But: Nvm, doing this is quite questionable, keep it :D